asmcomp/power/emit.mlp

```diff
@@ -527,6 +535,39 @@ let emit_alloc env i bytes dbginfo far =
     `{emit_label lbl}:	addi	{emit_reg i.res.(0)}, 31, {emit_int size_addr}\n`
   end
 
+let emit_poll env i return_label far =
+  if env.call_gc_label = 0 then env.call_gc_label <- new_label ();
+  let offset = Domainstate.(idx_of_field Domain_young_limit) * 8 in
+  `	{emit_string lg}	0, {emit_int offset}(30)\n`;
+  `	{emit_string cmplg}	31, 0\n`;
+  if not far then begin
+    begin match return_label with
+    | None ->
+    begin
+      `	bltl	{emit_label env.call_gc_label}\n`;
+      record_frame env i.live (Dbg_alloc [])
+    end
+    | Some return_label ->
+      begin
+        ` bge {emit_label return_label}\n`;
+        ` bl  {emit_label env.call_gc_label}\n`;
+        record_frame env i.live (Dbg_alloc []);
+        ` b   {emit_label return_label}\n`
```

Comment:
This looks less efficient than just
```
    bltl   env.call_gc_label
record_frame
    b return_label
```
(3 instructions instead of 2).

Reply:
I was pondering that as I wrote it. It's either one extra branch per loop iteration or one extra instruction added per loop.

Is the former a better trade on power?